### PR TITLE
add task11 with read write lock

### PR DESCRIPTION
**Pull request by VasylKiyko**
Hope it is good one, i found an example end try create my own RWLock

### CODE REVIEW

File: lab11.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+from contextlib import contextmanager
+from threading import Lock, Thread
+DATA = {'old_item_0': 0, 'old_item_1': 1, 'old_item_2': 2, 'old_item_3': 3}
+
+
+class RWLock:
+    def __init__(self):
+        self.write_lock = Lock()
+        self.read_lock = Lock()
+        self.readers = 0
```
Comment (polyakovartem) [inline lab11.py:10]: these variables should be declared private

File: lab11.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    def acquire_read(self):
+        self.read_lock.acquire()
+        try:
+            self.readers += 1
+            # як тільки з'являєтсья перший читач, блокуємо write lock
+            if self.readers == 1:
+                self.write_lock.acquire()
+        finally:
+            self.read_lock.release()
```
Comment (polyakovartem) [inline lab11.py:20]: the algorithm here does not solve anything. None of the reading threads should be blocked unless there is a writer thread writing. In your approach writer thread will be deadlocked by readers

File: lab11.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        DATA[thread_key] = thread_id
+
+
+if __name__ == '__main__':
+    lock = RWLock()
+
+    for x in range(5):
+        thread = Thread(target=write_data, args=(lock, x))
+        thread.start()
+        thread.join()
```
Comment (polyakovartem) [inline lab11.py:75]: the join method hear means that at any point of time you'll gonna have no more than two threads in the process, the main thread and/or the writer/reader thread

**Review by polyakovartem — CHANGES_REQUESTED** [review]

File: lab11.py (hunk 2 above)
Comment (VasylKiyko) [reply lab11.py:20]: but, here I thought, if someone read, anyone else cann't write in the same moment, if reader come earlier then writer. Is it bad?

Comment (VasylKiyko) [reply lab11.py:20]: Also, if I understand correct, I don't need read_lock at all?

Comment (polyakovartem) [reply lab11.py:20]: please review the existing implementation of RW locks in other languages

**Pull request closed without merging**